src/AvaloniaEdit/Editing/TextArea.cs

```diff
@@ -1200,7 +1200,9 @@ public override TextSelection Selection
                 }
                 set
                 {
-                    var selection = _textArea.Selection;
+                    if (_textArea == null) return;
+                    var selection =  _textArea.Selection;
+                    if (selection.StartPosition.Line == 0) return;
```

Comment:
Why do we need to return here?

Reply:
If not, it's a null reference exception, while accessing the selection.

Reply:
In `Text.Documnet.GetLineByNumber(int number)`, `0` is not supported.
https://github.com/AvaloniaUI/AvaloniaEdit/blob/be8fc934ba87692bc3d4a2e5360ae595c115b5f8/src/AvaloniaEdit/Document/TextDocument.cs#L939